src/app/api/api.actions.ts

```diff
@@ -0,0 +1,15 @@
+import { Action, createAction } from '@ngrx/store';
+import { HttpErrorResponse } from '@angular/common/http';
+import { of, OperatorFunction } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+// tslint:disable-next-line:typedef
+export function createApiErrorAction(type: string, errorMessageKey?: string) {
+  return createAction(type, (error: HttpErrorResponse) => ({ error, errorMessageKey }));
+}
+
+export type ApiErrorActionCreator = ReturnType<typeof createApiErrorAction>;
+
+export function catchApiError(apiErrorActionCreator: ApiErrorActionCreator): OperatorFunction<Action, Action> {
+  return catchError((error: HttpErrorResponse) => of(apiErrorActionCreator(error)));
+}
```

Comment:
Tu tez potrzebuje wyjasnienia xD

Reply:
Obudowałem rxjs'owy operator catchError we własny, żebyśmy nie musieli za każdym razem pisać tego samego kodu w effectach. Więcej mogę Ci wyjaśnić ustnie